lib/protocol/http1/connection.rb

```diff
@@ -135,7 +135,7 @@ def write_headers(headers)
 			end
 			
 			def read_line?
-				@stream.gets(CRLF, chomp: true)
+				@stream.gets(CRLF).chomp
```

Comment:
The problem with this is it allocates extra strings and this breaks memory usage assumptions. I'll need to confirm this is going to work in light of the memory limit specs we have.

Reply:
I checked and result is here https://gist.github.com/ganmacs/52fbcabdc1311d43dcd730442672a7a9.

> The problem with this is it allocates extra strings and this breaks memory usage assumptions

it's certainly so but, `@stream.gets(CRLF, chomp: true)` creates a new Hash instance and it allocates more memory than `@stream.gets(CRLF).chomp` ver.

Reply:
Yes, you are right, at the time I considered this.

Fortunately, it only allocates a hash when @stream is actually a Ruby IO instance. However, for `Async::IO::Stream` because `gets` is pure Ruby, it doesn't allocate a hash. It's very tricky to get right and I agree it's kind of crappy edge case.

I'm happy to work on supporting 2.3 as a baseline, but I just need to review how the entire thing fits together. It might even be the case that we just version detect, e.g. `if RUBY_VERSION <= "2.3" use .chomp`

Reply:
In pure ruby implementation of streams, `@stream.gets(CRLF, chomp: true)`  is more efficient too.

Reply:
>  e.g. if RUBY_VERSION <= "2.3" use .chomp

if you are okay with it. i'm ok too 👍 
https://github.com/socketry/protocol-http1/pull/1/commits/43802f07a896fe49b84c1ed08189502e50635ca6